rust/lance-core/src/datatypes/schema.rs

```diff
@@ -135,7 +135,7 @@ impl Schema {
     ///
     /// This is a DFS traversal where the parent is visited
     /// before its children
-    fn fields_pre_order(&self) -> SchemaFieldIterPreOrder {
+    pub fn fields_pre_order(&self) -> impl Iterator<Item = &Field> {
```

Comment:
By making the return type `impl Iterator` we can make this method public without making the returned struct public :)